tests/kernel/arm_irq_vector_table/src/arm_irq_vector_table.c

```diff
@@ -10,7 +10,59 @@
 #include <linker/sections.h>
 
 
-struct k_sem sem[3];
+/*
+ * Offset (starting from the beginning of the vector table)
+ * of the location where the ISRs will be manually installed.
+ */
+#define _ISR_OFFSET 0
+
+#if defined(CONFIG_SOC_SERIES_NRF52X)
+/* The customized solution for nRF52X-based platforms
+ * requires that the POWER_CLOCK_IRQn line equals 0.
+ */
+BUILD_ASSERT_MSG(POWER_CLOCK_IRQn == 0,
+	"POWER_CLOCK_IRQn != 0. Consider rework manual vector table.");
+
+/* The customized solution for nRF52X-based platforms
+ * requires that the RTC1 IRQ line equals 17 and is larger
+ * than the CONFIG_NUM_IRQS.
+ */
+BUILD_ASSERT_MSG(RTC1_IRQn == 17,
+	 "RTC1_IRQn != 17. Consider rework manual vector table.");
+BUILD_ASSERT_MSG(RTC1_IRQn >= CONFIG_NUM_IRQS,
+	 "RTC1_IRQn < NUM_IRQs. Consider rework manual vector table.");
```

Comment:
I still see no point in checking this. You can set `NUM_IRQS` in Kconfig, but reasonable values for this option are <= 3. For higher ones, the test won't work anyway, although they can still pass this check.

Reply:
Removed. I agree, not much sense on this one.